### ISSUE TITLE
I think I lost funds due to a bug

### ISSUE THREAD

**Issue by goldenNyan** [asgardex/asgard-exchange#390]
Hi awesome community,

I lost my funds due to a bug which started with Asgardex cascading with THORSwap. Here's what happened:

I've been an LP on BEPSwap and moved my funds to MCCN with the latest cap raise

1. I moved my bep2.rune and bep2.eth to bnb187ag8gxc3773lahj48k28h45zpc2ya9pwutvuu (Asgardex)

2. upgraded to native rune to thor12e8dn0dtuuk9mpmj9hqf4tvzw42zxeanu2p40c (Asgardex)

3. added liquidity to bep2.eth pool (Asgardex)

4. only rune went through ASYMMETRICALLY with a massive slippage (Asgardex)
<img width="793" alt="Screen Shot 2021-05-26 at 06 49 03" src="https://user-images.githubusercontent.com/84806162/119552203-8ec3e100-bdee-11eb-95e9-f9f038abfc7f.png">

5. bep2.ETH got stuck as below (Asgardex)
<img width="1577" alt="Screen Shot 2021-05-26 at 04 14 49" src="https://user-images.githubusercontent.com/84806162/119549904-d432df00-bdeb-11eb-9466-a26354f16c36.png">

6. received help from THORSwap community and managed to withdraw the pending funds (THORSwap)

7. funds returned as 50% rune and 50% bep2.eth (THORSwap)

8. now the first half of the liquidity I added got stuck as below (Asgardex & THORSwap)
<img width="1644" alt="Screen Shot 2021-05-26 at 05 55 27" src="https://user-images.githubusercontent.com/84806162/119550436-85d21000-bdec-11eb-8a9d-0e11cb627234.png">

9. received help from THORSwap and completed the LP (THORSwap)

10. but now native rune recovered from step 7 was withdrawn (instead from the initial fund that was swapped ASYMMETRICALLY against my wish)

11. I lost my funds (unintended asymmetrical swap & around 25% of the funds that I started wih)

I think the vault could be insolvent with these transactions and I might have lost my rune.

Please help...

UPDATE:

it looks like step 9 went through, but as another asymmetrical swap with a massive slippage. So I now have that amount minus the massive slippage. Now this particular process completely drained my rune and I am now again stuck in the network.

<img width="793" alt="Screen Shot 2021-05-26 at 08 17 58" src="https://user-images.githubusercontent.com/84806162/119563394-96d64d80-bdfb-11eb-8cac-38093f0c44fd.png">

UPDATE 2:
I gave another go at adding liquidity (using THORSwap), and it went smoothly this time. After doing some calculations, I don't think there should be an issue with regards to solvency. It looks like I've just lost a bunch of money due to a bug that led me to add liquidity asymmetrically several times.

**Comment by abcrypto30** [comment]
Facing a similar issue when I tried to add liquidity to the BNB-RUNE pool. 

https://explorer.binance.org/address/bnb1vlzve2kn8tncdjtxqxyg4aeyh33ycl0qskeupw


1. Converted my RUNE to native RUNE.

2. Tried to add 50-50 to BNB-RUNE pool.

3. Asgardex showed an error message "Insufficient balance"

4. However my BNB balance got deducted (RUNE still in my wallet) and Thorchain explorer does not show the transaction.

5. But my BNB address shows the outgoing BNB transaction as successful.

Is there anyway I can recover my funds?

**Comment by stuartwk** [comment]
Hey @goldenNyan, it looks like for some reason the xchain RUNE deposit failed, leaving your funds pending.
**When this happens, you can always hit https://app.asgard.exchange/deposit-sym-recovery to check for stuck funds and withdraw.** Keep in mind, withdrawing stuck funds will withdraw 100% from that pool, including previous deposits.

Could you share what bug you're referring to with multiple asymmetric deposits? Asgardex does not support asym deposits at the moment. Asym deposits are supported on THORSwap. Since the RUNE tx failed, it was stuck as pending since it was waiting on the symmetric RUNE deposit.

You can look at your tx history here -> https://thorchain.net/#/address/bnb187ag8gxc3773lahj48k28h45zpc2ya9pwutvuu. I see that you deposited 14k RUNE with only 0.002 ETH as a sym deposit, which would probably throw off the price of ETH in the system, and then followed that with a withdraw.

**Comment by goldenNyan** [comment]
Hi thank you for paying attention to my issue. The potential bug that I’m referring is the unintentional asymmetrical liquidity provision. The 14K rune + 0.002 eth transaction showed up as 14K rune + 62 ETH on asgardex and that is what I signed using my keystore. After it got approved only rune went through, stayed as pending. Tthorswap community confirmed that it went through as asymmetrical addition. Then the half got stuck, I recovered it, the initial was still stuck, completed the transaction another liquidity provision as performed asymmetrically.

again the issue that I’m referring to is that it showed up as asymmetrical provision on my end but only rune was added with the rest of the funds hanging. I started with 320k and ended with 280k.

**Comment by stuartwk** [comment]
@goldenNyan are you using the desktop app or the website app.asgard.exchange?

I ask because the memos you have look different than the memos we use on the web app. The desktop is a different codebase.
BNB.ETH tx -> https://explorer.binance.org/tx/6F6A848AD4641E240EF1A3752265A50697BF98586EF3E949D1492A90CE162B3A
THOR.RUNE tx -> https://thornode.thorchain.info/thorchain/tx/642AD6C8589495E16BFFAD02A4F110833622F9D696B81F0747DD8EC2EDEE92C9

You can see that the memos both start with 'ADD', but on the webapp we use '+'.
You can take a look https://github.com/asgardex/asgard-exchange/blob/master/src/app/_services/keystore-deposit.service.ts#L114 (BNB deposit) and https://github.com/asgardex/asgard-exchange/blob/master/src/app/deposit/confirm-deposit-modal/confirm-deposit-modal.component.ts#L211 (THOR.RUNE deposit).

**Comment by goldenNyan** [comment]
I used the desktop app. I don't trust myself to write memos correctly..

**Comment by stuartwk** [comment]
I see, that is actually a different codebase by a different team, which you can find at https://github.com/thorchain/asgardex-electron. Feel free to open a ticket there.

**Comment by goldenNyan** [comment]
Oh I see thank you for letting me know.

**Comment by stuartwk** [comment]
No problem, going to close this issue out. Feel free to raise another issue if it's regarding the web app 👍
